app/models/group.rb

```diff
@@ -3,7 +3,12 @@ class Group < ApplicationRecord
 
   has_many :group_admins, dependent: :destroy
   has_many :group_associations
-  has_many :users, through: :group_associations
+  has_many :users, -> do
```

Comment:
I'd say we should refrain from this approach.
gate has tendency towards 'read', this will put extra burden to the db.
Instead we should have periodic worker that check for expired users and automatically mark them inactive / remove them from group

Reply:
another problem with this approach is, you forcefully put additional (business-related) logic in this relation. You will have problem if you want to use this relation for different purpose